nexus-and-alo/nexus-strawman.md

```diff
@@ -0,0 +1,361 @@
+
+- Making external RPC calls or other I/O operations.
+- Using a full-fledged Temporal client for kicking off Workflows or interacting with other Temporal resources.
+- Interacting with the originating Request through a formal Response.
+
+### Nexus Task Worker
+
+In order to make forward progress on incoming Nexus Tasks, a valid Nexus Worker for the Nexus Instance must be registered.
+
+// TODO (Chad): improve if needed
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+
+	"github.com/cretz/nexus-poc/sdk/go/nexus/backend/client"
+	"github.com/cretz/nexus-poc/sdk/go/nexus/backend/worker"
+	"github.com/cretz/nexus-poc/sdk/temporal-go/temporalnexus"
+	temporalclient "go.temporal.io/sdk/client"
+	temporalworker "go.temporal.io/sdk/worker"
+)
+
+func main() {
+	ctx := context.TODO()
+
+	// Create client
+	client, err := client.Dial(ctx, client.Options{Target: "nexus-backend.example.com"})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	// Build worker with Temporal pieces
+	temporalClient, err := temporalclient.Dial(temporalclient.Options{HostPort: "nexus-backend.example.com:7233"})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer temporalClient.Close()
+	builder, err := temporalnexus.NewWorkerBuilder(temporalnexus.WorkerBuilderOptions{Client: temporalClient})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Add workflow
+	err = builder.AddWorkflowALOHandler("foo", temporalnexus.WorkflowALOHandlerOptions{
+		Workflow: SayHelloWorkflow,
+		Options:  temporalclient.StartWorkflowOptions{TaskQueue: "my-task-queue"},
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Add queries and signals
+	err = builder.AddQuerySyncHandler("foo/get", temporalnexus.QuerySyncHandlerOptions{
+		Query: "get-foo",
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = builder.AddSignalSyncHandler("foo/update-prefix", temporalnexus.SignalSyncHandlerOptions{
+		Signal: "update-prefix",
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = builder.AddSignalSyncHandler("foo/finish", temporalnexus.SignalSyncHandlerOptions{
+		Signal: "finish-workflow",
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Start Temporal worker
+	temporalWorker := temporalworker.New(temporalClient, "my-task-queue", temporalworker.Options{})
+	temporalWorker.RegisterWorkflow(SayHelloWorkflow)
+	if err := temporalWorker.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer temporalWorker.Stop()
+
+	// Start Nexus worker
+	worker, err := worker.New(worker.Options{Client: client, Service: "my-service", Handler: builder.BuildHandler()})
+	if err != nil {
+		log.Fatal(err)
+	} else if err = worker.Start(ctx); err != nil {
+		log.Fatal(err)
+	}
+	defer worker.Stop()
+
+	// Wait for completion
+	log.Print("Worker started, ctrl+c to stop")
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt)
+	<-signalCh
+}
+```
+
+### Calling Nexus from Temporal
```

Comment:
It seems a bit counter intuitive but maybe move this sample in front of the other, as this is something we all understand already and it helps explain what's going on with the Nexus handler example